src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -13,6 +18,8 @@
 	private DriverCommand currentCommand = null;
 
 	private Publisher changePublisher = new Publisher();
+	ArrayList<HistoryCommandItem> commandsList = new ArrayList<>();
+	int index;
```

Comment:
SRP: please extract history (data and basic behavior) to a separate class.
BTW: should by private